src/dd/args.rs

```diff
@@ -0,0 +1,412 @@
+use std::env;
+use std::sync::{Arc, Mutex};
+
+use super::util::{block, def, err, unblock,};
+use super::defs::{Arg, Config};
+
+pub fn parse_args() -> Arc<Mutex<Config>> {
+    let mut cf = Box::new(Config::default());
```

Comment:
I can't use mere stack variable because it needs to be modified both in parse_args and handlers.
So I use box to pass ownership to handler and back via return value from handler.

Yes, it's cumbersome, but otherwise one should resort to unsafe code as it's not possible to get two mutable links to one object.

Reply:
Why can't you pass it as a mutable reference to the handlers?